### ISSUE TITLE
Dataset Version - Missing summary and view differences details

### ISSUE THREAD

**Issue by lubitchv** [IQSS/dataverse#5510]
There is a bug in "Viewing Differences" for dataset metadata.
The bug can be seen here [https://demo.dataverse.org/dataset.xhtml?persistentId=doi%3A10.5072%2FFK2%2FAFDHNP]( https://demo.dataverse.org/dataset.xhtml?persistentId=doi%3A10.5072%2FFK2%2FAFDHNP).
In  Versions tab select 1.0 and 3.0 and press "View Differences". There will be no differences, although it should have differences. 

Steps to reproduce the bug:
1) Add new dataset with ssis file. Publish it.
2) Select file then Edit Files -> Metadata. Change File Metadata description, save it. Publish it.
3) In Metadata tab, Press "Add + Edit Metadata". Edit the fields (subtitle, alternative title, etc). Save it and publish
4) In Versions Tab select version 1 and version 3 and press "View Differences". One will not be able to see the differences, it is empty.

**Comment by pdurbin** [comment]
@lubitchv thanks for the bug report.

For the record, the demo server is running 3721306 (version 4.11). Here's a screenshot:

<img width="1361" alt="Screen Shot 2019-03-27 at 8 37 17 PM" src="https://user-images.githubusercontent.com/21006/55121309-56220300-50d0-11e9-82c4-a3be105a5804.png">

**Comment by mheppler** [comment]
Two other missing version details related issues to consider and complete or close are, _Versions Tab: Adding/removing a guestbook does not appear in version differences #4905_, as well as _Version Differences Detail - Tabular tags applied in DRAFT version not recorded in Version Details #3430_.

Was able to replicate this bug locally on develop (v 4.15). Just editing the file metadata description resulted in an empty popup.

<img width="609" alt="Screen Shot 2019-06-20 at 10 29 18 AM" src="https://user-images.githubusercontent.com/687227/59857583-3d2f9380-9347-11e9-9eb9-a07adaa83416.png">

Enabling request access of a restricted file in the edit terms pg results in an empty summary in the versions table on the dataset pg.

<img width="731" alt="Screen Shot 2019-06-20 at 10 29 03 AM" src="https://user-images.githubusercontent.com/687227/59857591-40c31a80-9347-11e9-9bfd-6b6563e858e1.png">

**Comment by donsizemore** [comment]
We hit what I believe to be this bug. Adding from server.log:

```
javax.el.ELException: /dataset.xhtml @1027,197 rendered="#{fileDiff.fileRest1 != null}":

at edu.harvard.iq.dataverse.DatasetVersionDifference$datasetFileDifferenceItem.getFileRest1(DatasetVersionDifference.java:1622)
```

**Comment by pdurbin** [comment]
@lubitchv hi! There's nothing in code review or QA right now so I'm taking a look at what's in "community dev" and noticed this is assigned to you. Are you still working on this issue? Should it be assigned to you? Thanks!

**Comment by lubitchv** [comment]
Hi Philip. No, I never worked on this issue. I just reported the bug that I saw. I do not think it should be assigned to me.

**Comment by pdurbin** [comment]
@lubitchv thanks! I just unassigned you can removed it from "community dev" (and from the board entirely). This undoes what @djbrooke did so heads up to him. 😄 Thanks for reporting the bug!

**Comment by mheppler** [comment]
Here is an expanded outline of the related issues reporting missing details from the versions, as I mentioned in my previous comment above.

- Dataset Versions Tab: Clicking on details link (diff between 2 vers) when only change is file replace shows nothing, log error. #6298
- Versions Tab: Adding/removing a guestbook does not appear in version differences #4905
- Dataset Version Tab: Restricting a file does not display filename in differences view, only id and md5, #3517
- Versions Tab: Enable request access value does not appear in version differences. #2432

**Comment by scolapasta** [comment]
Here's my 2 cents:

There are 3 categories of things here:
1. A bug with existing logic
2. Metadata missing from version differences (e.g. tags, request access)
3. (out of scope) handling changes that are not versioned (e.g. guestbook)

#3 should definitely be considered out of scope here, because we don't actually store guestbook info as versioned. And we need to decide if the change is to version it or to allow you to make changes without having to create a new version.

I think solving the bug and adding the missing metadata could be handled together since it is the same area of code.

**Comment by scolapasta** [comment]
@lubitchv what is an ssis file? (or was that a typo)

I just tried to reproduce on 4.20 with an image file and could not (with the steps from the original description).

**Comment by lubitchv** [comment]
Hi @scolapasta Sorry for the delay, I was off for several weeks and just returned to work yesterday. Yes, "ssis" is typo and it should be "spss". I tried to reproduced in demo (ver 4.20) and it seems to be fixed. [https://demo.dataverse.org/dataset.xhtml?persistentId=doi%3A10.70122%2FFK2%2FILMX3F](https://demo.dataverse.org/dataset.xhtml?persistentId=doi%3A10.70122%2FFK2%2FILMX3F) I just tried it in our dev environment that has version 4.17 and in that version it has this bug. So I guess in 4.20 it is fixed.
